src/Network/Algorand/Definitions.hs

```diff
@@ -0,0 +1,105 @@
+-- SPDX-FileCopyrightText: 2021 Serokell <https://serokell.io>
+--
+-- SPDX-License-Identifier: MPL-2.0
+
+-- | Definitions to work with Algo, grouped in this module for convenience.
+module Network.Algorand.Definitions
+  ( Network (..)
+  , NetworkError (..)
+  , networkParser
+
+  , Host
+  , DefaultHost (..)
+  , getDefaultHost
+  ) where
+
+import qualified Data.Map.Strict as Map
+import qualified Data.Text as T
+
+import Control.Exception.Safe (Exception)
+import Data.Aeson (FromJSON (..), ToJSON (..))
+import Data.Functor ((<&>))
+import Data.Map.Strict (Map)
+import Data.Text (Text)
+
+-- | Supported networks.
+data Network
+  = MainnetV1
+  | TestnetV1
+  | BetanetV1
+  deriving (Eq, Ord)
+
+-- | Mapping from `Network` to it's name
+networkToName :: Network -> Text
+networkToName = \case
+  MainnetV1 -> "mainnet-v1.0"
+  TestnetV1 -> "testnet-v1.0"
+  BetanetV1 -> "betanet-v1.0"
+
+networkParser :: String -> Either String Network
```

Comment:
Can you please change the argument to `Text`? We typically use `Text` by default. I see that here (in this package) you pass `networkParser` to `eitherReader` which works with `String`, but externally it may be used with `Text` (and I actually need to pass `Text` here).

Reply:
I used String here, because there were some transformations form string to text and there back from text to string (don't remember where exactly), but I can do it

Reply:
also `eitherReader` expects `String -> Either String Network`

Reply:
@gromakovsky please, take a look